src/Build/BackEnd/BuildManager/BuildParameters.cs

```diff
@@ -763,6 +768,15 @@ public string OutputResultsCacheFile
             set => _outputResultsCacheFile = value;
         }
 
+        /// <summary>
+        /// Track the environment variables actually read and used.
+        /// </summary>
+        public bool TrackReadsOnEnvironmentVariables
+        {
+            get => GetStaticBoolVariableOrDefault("MSBUILDTRACKENVVARREADS", ref s_trackReadsOnEnvironmentVariables, false);
```

Comment:
>GetStaticBoolVariableOrDefault("MSBUILDTRACKENVVARREADS", ref s_tr [](start = 19, length = 66)

should we expose this as part of the public API of MSBuild, instead of using an env var?
My rationale is: asking to track env vars from a command line execution of MSBuild never makes sense, and this is likely only going to be used by the static graph APIs. And an explicit flag in an API is always cleaner and more explicit than an env var

Reply:
Per our discussion in last week's meeting. It is expected that environment variables are used to switch on/off rare or quirky behaviors in msbuild. I've changed where this is implemented though from BuildParameters to Traits.